packages/daimo-api/src/contract/SwapClogMatcher.ts

```diff
@@ -20,39 +23,56 @@ export class SwapClogMatcher {
   // Given a set transaction hashes, load any transfers that are associated
   // with those transactions.
   async loadSwapTransfers(
-    pg: Pool,
+    kdb: Kysely<IndexDB>,
     from: number,
     to: number,
     chainId: number,
-    transactionHashes: string[]
+    blockNums: Set<bigint>,
+    txHashes: Set<Hex>
   ) {
     const startTime = Date.now();
 
-    const txHashes = transactionHashes.map((tx) => `\\x${tx.substring(2)}`);
+    const blockList = [...blockNums].sort().map((bn) => bn.toString());
+    const txList = [...txHashes].map((tx) => Buffer.from(hexToBytes(tx)));
 
     const result = await retryBackoff(
       `swapClogMatcher-logs-query-${from}-${to}`,
-      async () => {
-        return pg.query(
-          `SELECT * FROM erc20_transfers WHERE tx_hash = ANY($1::bytea[]);`,
-          [txHashes]
-        );
-      }
+      () =>
+        kdb
+          .selectFrom("index.daimo_transfer")
+          .select([
+            "block_hash",
+            "block_num",
+            "tx_hash",
+            "tx_idx",
+            "sort_idx",
+            "token",
+            "f",
+            "t",
+            "amount",
+          ])
+          .where("chain_id", "=", "" + chainConfig.chainL2.id)
+          .where("block_num", "in", blockList)
+          .where("tx_hash", "in", txList)
+          .execute()
     );
 
-    for (const row of result.rows) {
+    for (const row of result) {
       const log = {
         blockHash: bytesToHex(row.block_hash, { size: 32 }),
         blockNumber: BigInt(row.block_num),
         transactionHash: bytesToHex(row.tx_hash, { size: 32 }),
-        transactionIndex: row.tx_idx,
-        logIndex: row.log_idx,
-        address: getAddress(bytesToHex(row.log_addr, { size: 20 })),
+        transactionIndex: Number(row.tx_idx),
+        logIndex: Number(row.sort_idx),
+        address:
+          row.token == null
```

Comment:
in s^2, should we fill the token to zero address instead of having it as null?

Reply:
good call, S^2 does that now